propolis/src/inventory.rs

```diff
@@ -435,28 +391,19 @@ impl Record {
         &self.name
     }
 
-    fn first_child(&self) -> Option<EntityID> {
-        self.children.iter().next().copied()
+    pub fn parent(&self) -> Option<EntityID> {
+        self.parent
     }
 }
 
-pub trait Block {}
-pub trait Pci {}
-pub trait Chipset {}
-pub trait CharDev {}
-
 pub trait Entity: Send + Sync + 'static {
-    fn as_block(&self) -> Option<&dyn Block> {
-        None
-    }
-    fn as_pci(&self) -> Option<&dyn Pci> {
-        None
-    }
-    fn as_chipset(&self) -> Option<&dyn Chipset> {
-        None
-    }
-    fn as_chardev(&self) -> Option<&dyn CharDev> {
-        None
+    #[allow(unused_variables)]
+    fn state_transition(
```

Comment:
(definitely not blocking, but just for discussion)

The current expectation with this API is that we'l give entities a single blocking moment in time to react to state transitions, right?

Do we view it as limiting that the reaction to transitions is effectively serialized and synchronous? I'm wondering if at some point we'd want to be able to communicate to devices that an event is happening - like "migration start" - and let multiple devices begin acting on that event concurrently.

(Admittedly, this would also be possible with the current API, just with more states)